OutOfSchool/OutOfSchool.BusinessLogic/Services/Database/WorkshopService.cs

```diff
@@ -899,4 +893,29 @@ private async Task UpdateWorkshop()
     }
 
     private void ValidateOffsetFilter(OffsetFilter offsetFilter) => ModelValidationHelper.ValidateOffsetFilter(offsetFilter);
+
+    private async Task UpdateWorkshopStatusBySeatsLimitAndAvailability(uint newAvailableSeats, Workshop currentWorkshop)
+    {
+        var currentWorkshopTakenSeats = currentWorkshop.Applications.TakenSeats();
+
+        if (newAvailableSeats == uint.MaxValue
+            && currentWorkshop.AvailableSeats == currentWorkshopTakenSeats)
+        {
+            await UpdateStatus(new()
+            {
+                WorkshopId = currentWorkshop.Id,
+                Status = WorkshopStatus.Open,
+            }).ConfigureAwait(false);
+        }
+
+        if (newAvailableSeats < uint.MaxValue
+            && newAvailableSeats == currentWorkshopTakenSeats)
+        {
+            await UpdateStatus(new()
+            {
+                WorkshopId = currentWorkshop.Id,
+                Status = WorkshopStatus.Closed,
+            }).ConfigureAwait(false);
+        }
```

Comment:
(Logic) What if `newAvailableSeats > currentWorkshopTakenSeats`? Workshop will not be closed, though there is "overlimit" of participants. Please consider changing condition in this `if`.

Reply:
If the admin switched the workshop to a limited number of seats `newAvailableSeats < uint.MaxValue` and at the same time set a new available number of seats greater than the current number of taken seats `newAvailableSeats > currentWorkshopTakenSeats`, we leave the current workshop status.

Reply:
I'm sorry, ofc I was asking about opppsite - `newAvailableSeats < currentWorkshopTakenSeats` case. 
If new available seats number are less than already taken we should at least block new registrations (so change status to Closed). 
Current code doesn't check this condition, instead only precise equality is checked `newAvailableSeats == currentWorkshopTakenSeats`, thus the status won't be set to `Closed` for the case when 5 seats are taken, but new available is 3.